src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -193,9 +192,7 @@ internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, Diagnos
                     {
                         generator.Initialize(pipelineContext);
                     }
-#pragma warning disable CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete; tracked by https://github.com/dotnet/roslyn/issues/58375
-                    catch (Exception e) when (FatalError.ReportIfNonFatalAndCatchUnlessCanceled(e, cancellationToken))
-#pragma warning restore CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete
+                    catch (Exception e) when (!ExceptionUtilities.IsCurrentOperationBeingCancelled(e, cancellationToken))
```

Comment:
Hmm, I'd imagine Initialize can't ever throw a cancellation exception, right? So should the filter just be removed here?

Reply:
Agreed. Changed